app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ReviewsController < ApplicationController
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @review = Review.new(review_params)
+    @beer = Beer.find(params[:beer_id])
+    if @review.save
+      flash[:notice] = "Review Saved"
+      redirect_to beer_path(@beer)
+    else
+      flash[:notice] = "Invalid Input"
+      render :new
+    end
+  end
+
+  def edit
+    @review = Review.find(params[:id])
+  end
+
+  def update
+    @review = Review.find(params[:id])
+    @beer = Beer.find(params[:beer_id])
```

Comment:
You don't need to find the beer explicitly. You can find the beer from the review object:

```ruby
@beer = @review.beer
```